internal/pkg/addons/addons.go

```diff
@@ -0,0 +1,78 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package addons contains the service to manage addons.
+package addons
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/workspace"
+)
+
+const (
+	paramsFileName  = "params"
+	outputsFileName = "outputs"
+	indent          = "  " // two spaces
+)
+
+type workspaceService interface {
+	ReadAddonsFile(appName, fileName string) ([]byte, error)
+	ListAddonsFiles(appName string) ([]string, error)
+}
+
+// Addons represent additional resources for an application.
+type Addons struct {
+	appName string
+
+	ws workspaceService
+}
+
+// NewAddons creates an Addons struct given an application name.
+func NewAddons(appName string) (*Addons, error) {
+	ws, err := workspace.New()
+	if err != nil {
+		return nil, fmt.Errorf("workspace cannot be created: %w", err)
+	}
+	return &Addons{
+		appName,
+		ws,
+	}, nil
+}
+
+// Template concatenates params.yml, policy.yml, {resource}.yml, and outputs.yml to generate
+// the CloudFormation template.
+func (a *Addons) Template() (string, error) {
+	addonFiles, err := a.ws.ListAddonsFiles(a.appName)
+	if err != nil {
+		return "", fmt.Errorf("list addon files: %w", err)
+	}
+	var resources, params, outputs string
+	for _, fileName := range addonFiles {
+		content, err := a.ws.ReadAddonsFile(a.appName, fileName)
+		if err != nil {
+			return "", fmt.Errorf("read addon file %s: %w", fileName, err)
+		}
+		switch name := strings.TrimSuffix(fileName, workspace.AddonsFileSuffix); name {
```

Comment:
maybe we shouldn't even trim and match strictly against "params.yml" and "outputs.yml"? I think that would simplify the code a little